Customizing/global/plugins/Services/Cron/CronHook/EffectivenessAnalysisReminder/classes/ilEffAnalysisActions.php

```diff
@@ -44,75 +44,114 @@ public function getOpenEffectivenessAnalysis($user_id) {
 	}
 
 	/**
-	 * Should send first reminder
+	 * Get user ids first mail should be send
 	 *
 	 * @param int 		$crs_id
 	 * @param int 		$superior_id
+	 * @param int[] 	$user_ids
 	 *
-	 * @return bool
+	 * @return int[]
 	 */
-	public function shouldSendFirstReminder($crs_id, $superior_id) {
-		$row_count = $this->db->getNumRowsOfSentFirstReminder($crs_id, $superior_id, self::FIRST);
+	public function getUserIdsForFirstMail($crs_id, $superior_id, $user_ids) {
+		$all_sent_user_ids = $this->db->getUserIdsTypeIsSend($crs_id, $superior_id, self::FIRST);
+
+		if(count($all_sent_user_ids) == 0) {
+			return $user_ids;
+		}
 
-		if($row_count == 0) {
-			return true;
+		$to_send = array();
+		foreach($all_sent_user_ids as $date => $sent_user_ids) {
+			$to_send = array_merge($to_send, array_diff($user_ids, $sent_user_ids, $to_send));
```

Comment:
Ich habe das dringende Gefühl, dass all das in der Datenbank besser aufgehoben wäre. Ich vermute, dass hängt auch mit dem `implode`-Vorschlag von unten zusammen.